docs/root/configuration/upstream/cluster_manager/cluster_stats.rst

```diff
@@ -69,7 +69,9 @@ Every cluster has a statistics tree rooted at *cluster.<name>.* with the followi
   upstream_rq_cancelled, Counter, Total requests cancelled before obtaining a connection pool connection
   upstream_rq_maintenance_mode, Counter, Total requests that resulted in an immediate 503 due to :ref:`maintenance mode<config_http_filters_router_runtime_maintenance_mode>`
   upstream_rq_timeout, Counter, Total requests that timed out waiting for a response
+  upstream_rq_timeout_budget_global, Histogram, What percentage of the global timeout was used waiting for a response
```

Comment:
Can you mention the idiosyncrasies around what happens when there is no timeout configured and what default values show up  when?

Reply:
If the timeout is set to 0/infinite, then the histogram will be filled with zeroes.  Otherwise, the time used out of the default timeout will be written.

Are you suggesting this be documented explicitly in this file?

Reply:
Previous comment is outdated.

Based on later feedback, this won't be set without the correct flag.  Should this be in its own table for clarity of documentation?